app/assets/javascripts/components/AccountMenu/ConfirmPassword.tsx

```diff
@@ -0,0 +1,172 @@
+              setPassword('');
+              handleGoBack();
+            });
+          })
+          .finally(() => {
+            setIsRegistering(false);
+          });
+      } else {
+        application.alertService
+          .alert(STRING_NON_MATCHING_PASSWORDS)
+          .finally(() => {
+            setConfirmPassword('');
+            passwordInputRef?.current.focus();
+          });
+      }
+    };
+
+    const handleGoBack = () => {
+      setMenuPane(AccountMenuPane.Register);
+    };
+
+    return (
+      <>
+        <div className="flex items-center px-3 mt-1 mb-3">
+          <IconButton
+            icon="arrow-left"
+            title="Go back"
+            className="flex mr-2 color-neutral"
+            onClick={handleGoBack}
+            focusable={true}
+            disabled={isRegistering}
+          />
+          <div className="sn-account-menu-headline">Confirm password</div>
+        </div>
+        <div className="px-3 mb-3 text-sm">
+          Because your notes are encrypted using your password,{' '}
+          <span className="color-dark-red">
+            Standard Notes does not have a password reset option
+          </span>
+          . If you forget your password, you will permanently lose access to
+          your data.
+        </div>
+        <form onSubmit={handleConfirmFormSubmit} className="px-3 mb-1">
+          <InputWithIcon
+            className="mb-2"
+            icon="password"
+            inputType={showPassword ? 'text' : 'password'}
+            placeholder="Confirm password"
+            value={confirmPassword}
+            onChange={handlePasswordChange}
+            onKeyDown={handleKeyDown}
+            toggle={{
+              toggleOnIcon: 'eye-off',
+              toggleOffIcon: 'eye',
+              title: 'Show password',
+              toggled: showPassword,
+              onClick: setShowPassword,
+            }}
+            ref={passwordInputRef}
+            disabled={isRegistering}
+          />
+          <Button
+            className="btn-w-full mt-1 mb-3"
+            label={
+              isRegistering ? 'Creating account...' : 'Create account & sign in'
+            }
+            type="primary"
+            onClick={handleConfirmFormSubmit}
+            disabled={isRegistering}
+          />
+          <Checkbox
+            name="is-ephemeral"
+            label="Stay signed in"
+            checked={!isEphemeral}
+            onChange={handleEphemeralChange}
+            disabled={isRegistering}
+          />
+          {notesAndTagsCount > 0 ? (
+            <Checkbox
+              name="should-merge-local"
+              label={`Merge local data (${notesAndTagsCount} notes and tags)`}
+              checked={shouldMergeLocal}
+              onChange={handleShouldMergeChange}
+              disabled={isRegistering}
+            />
+          ) : null}
+        </form>
+        <div className="h-1px my-2 bg-border"></div>
+        <AdvancedOptions
```

Comment:
We could have only once instance of the AdvanceOptions component in the account menu index, right below the MenuPaneSelector, rendering conditionally if `currentPane` is one of the panes that should include it (CreateAccount, SignIn, ConfirmPassword). That way, the value for the custom sync server checkbox for example would be preserved when navigating panes without having to store it in app state, and if a user expands the advanced options and then goes to a different pane the options will stay expanded. Wdyt?